server/server.go

```diff
@@ -835,14 +836,31 @@ func (server *Server) loadEntryPointConfig(entryPointName string, entryPoint *En
 	return rewrite, nil
 }
 
-func (server *Server) buildDefaultHTTPRouter() *mux.Router {
+func (server *Server) buildDefaultHTTPRouter(noRouteResponseCode int) *mux.Router {
 	router := mux.NewRouter()
-	router.NotFoundHandler = http.HandlerFunc(notFoundHandler)
+	router.NotFoundHandler = newNotFounderHandler(noRouteResponseCode)
 	router.StrictSlash(true)
 	router.SkipClean(true)
 	return router
 }
 
+func newNotFounderHandler(responseCode int) http.Handler {
+	if responseCode == 0 {
+		responseCode = http.StatusNotFound
+	}
+	if http.StatusText(responseCode) == "" {
+		log.Warnf("Wrong NoRouteResponseCode %d configured. Falling back to 404 (http.StatusNotFound)", responseCode)
```

Comment:
Thank you for the comment @discordianfish. I was already wondering whether this use-case exists. Indeed I was not very interested in the body itself that is returned, but used the `http.StatusText()` function as a kind of validation that the HTTP Status Code is a valid one. I was checking now the RFC and valid is basically everything in the range of 100 - 599.   See [here](https://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html#sec6.1.1).

Therefore I changed now the implementation to respect this use-case.